iis-wf/iis-wf-export-actionmanager/src/test/java/eu/dnetlib/iis/wf/export/actionmanager/module/DocumentToSoftwareUrlActionBuilderModuleFactoryTest.java

```diff
@@ -0,0 +1,138 @@
+    }
+
+    @Test(expected = TrustLevelThresholdExceededException.class)
+    public void test_build_below_threshold() throws Exception {
+        // given
+        DocumentToSoftwareUrls matchedOrgBelowThreshold = buildDocumentToSoftwareUrl(
+                docId, softwareUrl, repositoryName, 0.4f);
+        // execute
+        module.build(matchedOrgBelowThreshold, agent, actionSetId);
+    }
+
+    @Test
+    public void test_build() throws Exception {
+        // execute
+        List<AtomicAction> actions = module.build(documentToSoftwareUrl, agent, actionSetId);
+
+        // assert
+        assertNotNull(actions);
+        assertEquals(1, actions.size());
+        AtomicAction action = actions.get(0);
+        assertNotNull(action);
+        assertNotNull(action.getRowKey());
+        assertEquals(actionSetId, action.getRawSet());
+        assertEquals(docId, action.getTargetRowKey());
+        assertEquals(Type.result.toString(), action.getTargetColumnFamily());
+        assertOaf(action.getTargetValue());
+    }
+
+    // ----------------------- PRIVATE --------------------------
+
+    private static DocumentToSoftwareUrls buildDocumentToSoftwareUrl(String docId, String softUrl, 
+            String repositoryName, float confidenceLevel) {
+        DocumentToSoftwareUrls.Builder builder = DocumentToSoftwareUrls.newBuilder();
+        builder.setDocumentId(docId);
+        SoftwareUrl.Builder softBuilder = SoftwareUrl.newBuilder();
+        softBuilder.setSoftwareUrl(softUrl);
+        softBuilder.setRepositoryName(repositoryName);
+        softBuilder.setConfidenceLevel(confidenceLevel);
+        builder.setSoftwareUrls(Lists.newArrayList(softBuilder.build()));
+        return builder.build();
+    }
+
+    private void assertOaf(byte[] oafBytes) throws InvalidProtocolBufferException {
+        assertNotNull(oafBytes);
+        Oaf.Builder oafBuilder = Oaf.newBuilder();
+        oafBuilder.mergeFrom(oafBytes);
+        Oaf oaf = oafBuilder.build();
+        assertNotNull(oaf);
+
+        assertTrue(KindProtos.Kind.entity == oaf.getKind());
+        assertNotNull(oaf.getEntity());
+        assertEquals(docId, oaf.getEntity().getId());
+        assertNotNull(oaf.getEntity().getResult());
+        assertEquals(1, oaf.getEntity().getResult().getExternalReferenceList().size());
+        ExternalReference externalReference = oaf.getEntity().getResult().getExternalReferenceList().get(0);
+        assertNotNull(externalReference);
+        assertEquals(softwareUrl, externalReference.getUrl());
+        assertEquals(repositoryName, externalReference.getSitename());
+        
+        assertNotNull(externalReference.getQualifier());
+        assertNotNull(externalReference.getDataInfo());
+
+        float normalizedTrust = matchStrength * module.getConfidenceToTrustLevelNormalizationFactor();
+        assertEquals(normalizedTrust, Float.parseFloat(externalReference.getDataInfo().getTrust()), 0.0001);
```

Comment:
Thanks, good you spot that one. I'm on `en_US` so didn't notice that.

We need to be locale unaware, one way fix to this is instantiating `decimalFormat` differently:
```
this.decimalFormat = (DecimalFormat) NumberFormat.getInstance(Locale.ENGLISH);
this.decimalFormat.applyPattern("#.####");
```